sim/src/mechanics/walking.rs

```diff
@@ -932,3 +977,38 @@ impl IndexableKey for PedestrianID {
         self.0
     }
 }
+
+/// Returns a number in (0, 1] to multiply speed by to account for current crowdedness.
+///
+/// We could get really fancy here and slow people down only when they're part of a crowd, or
+/// passing people going the opposite direction. But start simple -- keep a fixed speed for the
+/// entire time on a sidewalk, and base the decision on how many people are there when entering the
+/// sidewalk.
+fn crowdedness_penalty(
+    map: &Map,
+    traversable: Traversable,
+    peds_per_traversable: &MultiMap<Traversable, PedestrianID>,
+) -> f64 {
+    // 1) How many people are in the space
+    let num_people = peds_per_traversable.get(traversable).len();
+    // 2) The length of the sidewalk or crosswalk
+    let len = traversable.get_polyline(map).length();
+    // 3) The width of the space
+    let width = match traversable {
+        // Sidewalk
+        Traversable::Lane(l) => map.get_l(l).width,
+        // For crosswalks, the thinner of the two sidewalks being connected
+        Traversable::Turn(t) => map.get_l(t.src).width.min(map.get_l(t.dst).width),
+    };
+
+    // Person per area, assuming everyone's equally spread out
```

Comment:
So a thin but really long sidewalk will seem OK, because the area will be high. In reality, people aren't evenly distributed along the length, of course.